app/protocol/flows/blockrelay/handle_relay_block_requests.go

```diff
@@ -27,26 +28,16 @@ func HandleRelayBlockRequests(context RelayBlockRequestsContext, incomingRoute *
 		getRelayBlocksMessage := message.(*appmessage.MsgRequestRelayBlocks)
 		for _, hash := range getRelayBlocksMessage.Hashes {
 			// Fetch the block from the database.
-			block, err := context.DAG().BlockByHash(hash)
-			if blockdag.IsNotInDAGErr(err) {
+			block, err := context.Domain().Consensus().GetBlock(hash)
+			if errors.Is(err, database.ErrNotFound) {
```

Comment:
Since we know it's possible to be requested a block that we don't have, please use GetBlockInfo instead to check whether it exists.